cmd/limactl/copy.go

```diff
@@ -128,13 +152,23 @@ func copyAction(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	}
+
+	var cmdArgs []string
 	sshArgs := sshutil.SSHArgsFromOpts(sshOpts)
+	if useRsync {
+		// for rsync, add the -e flag with SSH options
+		scpFlags = append(scpFlags, "-e", fmt.Sprintf("ssh %s", strings.Join(sshArgs, " ")))
+		cmdArgs = append(cmdArgs, append(scpFlags, scpArgs...)...)
+	} else {
+		// for scp, append SSH options and scpArgs
+		cmdArgs = append(cmdArgs, append(sshArgs, scpArgs...)...)
+	}
```

Comment:
This change complicates the code by adding useRsync condition on every step. This is not maintainable.

A better way to do this is to add a function to create a scp command arguments, and a function to create rsync command arguments. Then use shared code to run the command.